pkg/logger/logger.go

```diff
@@ -0,0 +1,87 @@
+package logger
+
+import (
+	"os"
+	"testing"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+// Logger is a minimal subset of smartcontractkit/chainlink/core/logger.Logger
+type Logger interface {
+	With(args ...interface{}) Logger
+	Named(name string) Logger
+
+	Debug(args ...interface{})
+	Info(args ...interface{})
+	Warn(args ...interface{})
+	Error(args ...interface{})
+	Panic(args ...interface{})
+	Fatal(args ...interface{})
+
+	Debugf(format string, values ...interface{})
+	Infof(format string, values ...interface{})
+	Warnf(format string, values ...interface{})
+	Errorf(format string, values ...interface{})
+	Panicf(format string, values ...interface{})
+	Fatalf(format string, values ...interface{})
+
+	Debugw(msg string, keysAndValues ...interface{})
+	Infow(msg string, keysAndValues ...interface{})
+	Warnw(msg string, keysAndValues ...interface{})
+	Errorw(msg string, keysAndValues ...interface{})
+	Panicw(msg string, keysAndValues ...interface{})
+	Fatalw(msg string, keysAndValues ...interface{})
+
+	Sync() error
```

Comment:
Can you add some documentation about when to use `Sync()` and what it does?

Reply:
Is pointing to core `logger.Logger` and `zap.SugaredLogger` from the type level comment sufficient?

Reply:
Yeah!